toolkits/google_sheets/arcade_google_sheets/tools/read.py

```diff
@@ -20,23 +21,39 @@
 async def get_spreadsheet(
     context: ToolContext,
     spreadsheet_id: Annotated[str, "The id of the spreadsheet to get"],
+    start_row: Annotated[int, "Starting row number (1-indexed, defaults to 1)"] = 1,
+    start_col: Annotated[
+        str, "Starting column letter(s) or 1-based column number (defaults to 'A')"
+    ] = "A",
+    max_rows: Annotated[
+        int,
+        "Maximum number of rows to fetch for each sheet in the spreadsheet. "
+        "Must be between 1 and 1000. Defaults to 1000.",
+    ] = 1000,
+    max_cols: Annotated[
+        int,
+        "Maximum number of columns to fetch for each sheet in the spreadsheet. "
+        "Must be between 1 and 26. Defaults to 26.",
```

Comment:
What an odd default!

Reply:
So I want I really wanted to do was remove the max_rows and max_cols input parameters, and then the tool itself determines the dimensions to fetch with preference for columns. Maybe the total number of cells can be configurable.

Examples: A 25row X 1000 column sheet would be supported by a single tool call. Same with a 10,000 row X 2 column sheet.